packages/sil-governance/src/routes/create-proposal/components/FormOptions/index.tsx

```diff
@@ -0,0 +1,36 @@
+import { ProposalType } from "@iov/bns-governance";
+import { FormApi } from "final-form";
+import React from "react";
+
+import AddCommitteeMember from "./AddCommitteeMember";
+import AddValidator from "./AddValidator";
+import AmendCommitteeQuorum from "./AmendCommitteeQuorum";
+import AmendCommitteeThreshold from "./AmendCommitteeThreshold";
+import AmendProtocol from "./AmendProtocol";
+import DistributeFunds from "./DistributeFunds";
+import ReleaseGuaranteeFunds from "./ReleaseGuaranteeFunds";
+import RemoveCommitteeMember from "./RemoveCommitteeMember";
+import RemoveValidator from "./RemoveValidator";
+
+interface Props {
+  form: FormApi;
+  proposalType: ProposalType;
+}
+
+const ProposalOptions = ({ form, proposalType }: Props): JSX.Element => {
+  return (
+    <React.Fragment>
+      {proposalType === ProposalType.AmendProtocol && <AmendProtocol form={form} />}
+      {proposalType === ProposalType.AddCommitteeMember && <AddCommitteeMember form={form} />}
+      {proposalType === ProposalType.RemoveCommitteeMember && <RemoveCommitteeMember form={form} />}
+      {proposalType === ProposalType.AmendElectionRuleThreshold && <AmendCommitteeThreshold form={form} />}
+      {proposalType === ProposalType.AmendElectionRuleQuorum && <AmendCommitteeQuorum form={form} />}
+      {proposalType === ProposalType.AddValidator && <AddValidator form={form} />}
+      {proposalType === ProposalType.RemoveValidator && <RemoveValidator form={form} />}
+      {proposalType === ProposalType.ReleaseGuaranteeFunds && <ReleaseGuaranteeFunds form={form} />}
+      {proposalType === ProposalType.DistributeFunds && <DistributeFunds form={form} />}
```

Comment:
Wouldn't a switch statement be clearer? This way I have to look down the whole list to work out that there'll only be one child component.

Reply:
Of course if you want you can use a switch, but in this case, I suggest to leave the code as it is because it is a very standard way to configure conditionally views on React. And if you use a switch it is going to look strange. 

But guys, if you definitely want to do a switch, feel free do it. Just my 2 cents.

Reply:
In general I'm totally on board with conditional views. But here I had to do some work to figure out that it's displaying *one* component dependent on the proposal type, whereas a `switch` would make this 100% clear at a glance.

Reply:
An alternative here would be a map and no conditional at all:
```tsx
const proposalMap = {
  [ProposalType.AddValidator]: AddValidator,
  [ProposalType.RemoveValidator]: RemoveValidator,
  // ...
};
// ...
const FormComponent = proposalMap[proposalType];
// ...
<FormComponent form={form} />
```

Reply:
I agree with both points of view. I like the map solution because:
1. It has just one line of code per case.
2. The return function has JSX code, which adds familiarity. (The current switch statement has a not needed `React.Fragment` that if omitted leaves `return getCurrentProposalOption()` which seems kind weird)

Reply:
The map option also makes it transparent that these components all take the same props.